rclpy/rclpy/context.py

```diff
@@ -149,3 +152,18 @@ def get_domain_id(self):
             raise RuntimeError('Context must be initialized before it can have a domain id')
         with self.__context, self._lock:
             return self.__context.get_domain_id()
+
+    def __enter__(self) -> "Context":
+        # We do not accept parameters here. If one wants to customize the init() call,
+        # they would have to call it manaully and not use the ContextManager convenience
+        self.init()
+        return self
+
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None,
```

Comment:
IIUC RHEL 9 and Windows still have Python < 3.10, so we'll need to use `Optional` here

* https://www.ros.org/reps/rep-2000.html#iron-irwini-may-2023-november-2024
* https://peps.python.org/pep-0604/

Reply:
Whhop, missed that.